### ISSUE TITLE
Crafting Recipes Loading Time.

### ISSUE THREAD

**Issue by Juafr** [CleverRaven/Cataclysm-DDA#54989]
### Describe the bug

I don't know if it is really a bug, but it is something I noticed in experimental and doesn't happen in stable.
When you set up a high item spawn number, like the max:10 the loading screen stops at crafting recipes for a lot longer than if you set the item spawn low, like the default 1.

### Steps to reproduce

1. Start a new world modless, set world options item spawn 10.
2. Count the time it gets to load the game past "crafting recipes".
3. Repeat step one with item spawn 1.
4. Repeat step two and compare the loading times.

### Expected behavior

The expected behavior is for items spawn setting not affecting the loading times.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 8f56778 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda]
]

**Comment by BrettDong** [comment]
The "Crafting Recipes" step displayed in loading UI performs a lot of temporary map generation, and a high spawn rate makes on-map item spawn to take longer CPU time.

**Comment by Juafr** [comment]
That explanation would work if it happened in stable too. It does not happen, even if it is a heavily modded stable 0.F-2 game there is no delay.
To clarify even further, to compensate for my poor english, there is no delay in the "Crafting Recipes" step in stable at all.

**Comment by BrettDong** [comment]
Yes because experimental has changed a lot.

**Comment by Juafr** [comment]
Well, it's a shame loading times have gotten so much worse.
I found out that changing the item spawn back to 1 in game before quitting reduces loading times when loading back. Then once inside the game you can change it back to 10 without problems, that way the only time you get long loading is during world creation.
If those high loading times are not a bug but a new characteristic feel free to close this issue.

**Comment by BrettDong** [comment]
This is a performance problem worth optimizing in the future.

**Comment by mqrause** [comment]
I did a bit of profiling with these settings:
![grafik](https://user-images.githubusercontent.com/38702195/176978518-b28a3585-6c57-4af9-a8b6-33b90f0b1bb7.png)

The result was pretty bad (90s on my slowish system, the rest of the loading was ~20s):
![grafik](https://user-images.githubusercontent.com/38702195/176978447-a33dc21e-2fe4-4d92-b2aa-df3ae9dfd550.png)

However this is mostly the overflow code checking if the items can overflow to some tile. I don't know too much about the json loading and mapgen code, but that doesn't sound like something that's all too relevant for recipe loading, so it can possibly be skipped?

**Comment by BrettDong** [comment]
Because item spawn is part of mapgen.

**Comment by mqrause** [comment]
Yeah, I meant specifically skipping the overflow check in `map::add_item_or_charges`, because that's where the 80% `map::route` come from. If that can be skipped, it'd reduce loading times with those settings by roughly two thirds. With a default world it would only be 4-5% time saved.

Though I gotta say I don't understand why item spawning is even happening there to such a degree? As far as I understand this is only due to faction camp blueprint requirements getting calculated. So it applies the mapgen update for all of the blueprints (~1700 apparently) on an empty map, but that shouldn't even include any items?
